### PR TITLE
[chore] expose codeowners in README

### PR DESCRIPTION
**Pull request by atoulme**
**Description:**
Add fields to status metadata and changes to README template to expose codeowners in component READMEs.

**Link to tracking Issue:**
https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23367

### CODE REVIEW

**Review by songy23 — APPROVED** [review]

File: cmd/mdatagen/templates/readme.md.tmpl [hunk 1]
```diff
@@ -19,6 +19,11 @@
 {{- end }}
 {{- if ne $class "" }}
 | Issues        | ![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3A{{ $class }}%2F{{ $shortName }}%20&label=open&color=orange&logo=opentelemetry) ![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3A{{ $class }}%2F{{ $shortName }}%20&label=closed&color=blue&logo=opentelemetry) |
+{{- end }}
+{{- if .Status.Codeowners }}
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | {{ stringsJoin .Status.Codeowners.Active ", " }} |
```
Comment (codeboten) [inline cmd/mdatagen/templates/readme.md.tmpl:24]: Should these be links to the code owners?

File: receiver/splunkhecreceiver/metadata.yaml [hunk 2]
```diff
@@ -5,3 +5,6 @@ status:
   stability:
     beta: [metrics, logs]
   distributions: [contrib, splunk, sumo]
+  codeowners:
+    active: ["@atoulme"]
```
Comment (codeboten) [inline receiver/splunkhecreceiver/metadata.yaml:9]: Why have the field here? Could the info be pulled from the codeowners file? mostly im just asking to have a single source of truth

Comment (atoulme) [reply receiver/splunkhecreceiver/metadata.yaml:9]: See discussion in https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/23367, we eventually will want to generate .github/CODEOWNERS from this metadata.

File: cmd/mdatagen/templates/readme.md.tmpl (hunk 1 above)
Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl:24]: Not sure what you mean - to the .github/CODEOWNERS file?

Comment (codeboten) [reply cmd/mdatagen/templates/readme.md.tmpl:24]: i meant to https://github.com/<username>, the @ symbol makes me think it was intended to be a link in the readme.

File: receiver/signalfxreceiver/metadata.yaml [hunk 3]
```diff
@@ -5,3 +5,6 @@ status:
   stability:
     beta: [metrics, logs]
   distributions: [contrib, splunk, sumo]
+  codeowners:
+    active: ["@dmitryax"]
+    emeritus:
```
Comment (codeboten) [inline receiver/signalfxreceiver/metadata.yaml:10]:
```suggestion
    emeritus:

```

File: cmd/mdatagen/templates/readme.md.tmpl (hunk 1 above)
Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl:24]: Now I see, yes, let me do that.

**Review by dmitryax — APPROVED** [review]

**Comment by dmitryax** [comment]
LGTM, but I think we should have one source of truth for code owners. So far, it has been `.github/CODEOWNERS`. If we are changing it, it'd be great to generate `.github/CODEOWNERS`

File: cmd/mdatagen/templates/readme.md.tmpl (hunk 1 above)
Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl]: Done!

**Comment by atoulme** [comment]
Yes, that's why I say in the issue, let me paste it here for clarification:
* [ ] Implement the changes to mdatagen, supporting new fields in status metadata and README. **(this PR)**
* [ ] Add the metadata to all components to the status section of metadata.yaml
* [ ] Generate the CODEOWNERS file from status metadata and check it in CI
* [ ] Change the documentation to offer to change code owners through the status metadata

**Comment by atoulme** [comment]
Rebasing off latest main just to make sure ; anything blocking merging this PR at this time?

File: cmd/mdatagen/templates/readme.md.tmpl [hunk 4]
```diff
@@ -20,6 +20,12 @@
 {{- if ne $class "" }}
 | Issues        | ![Open issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aopen%20label%3A{{ $class }}%2F{{ $shortName }}%20&label=open&color=orange&logo=opentelemetry) ![Closed issues](https://img.shields.io/github/issues-search/open-telemetry/opentelemetry-collector-contrib?query=is%3Aissue%20is%3Aclosed%20label%3A{{ $class }}%2F{{ $shortName }}%20&label=closed&color=blue&logo=opentelemetry) |
 {{- end }}
+{{- if .Status.Codeowners }}
+{{- $codeowners := userLinks .Status.Codeowners.Active }}
+{{- $emeritus := userLinks .Status.Codeowners.Emeritus }}
+| [Code Owners](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/main/CONTRIBUTING.md#becoming-a-code-owner)    | {{ stringsJoin $codeowners ", " }} |
+| Emeritus      | {{ stringsJoin $emeritus ", " }} |
```
Comment (dmitryax) [inline cmd/mdatagen/templates/readme.md.tmpl:27]: Maybe put a condition here not to generate the empty line if there are no Emeritus?

Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl:27]: It's not really an empty line, it shows:
```
| Emeritus |   |
```

Which indicates that there are no emeritus codeowners for this component. That in itself is a piece of information. We can also remove it entirely, in which case folks won't know we allow to record emeritus. I think I'm ok with the idea to remove that line if there are no emeritus code owners to keep the status table lean.

Comment (dmitryax) [reply cmd/mdatagen/templates/readme.md.tmpl:27]: I'd say it's cleaner without it, but feel free to keep it if you think otherwise

Comment (atoulme) [reply cmd/mdatagen/templates/readme.md.tmpl:28]: np, fixed.

**Pull request merged**